code/datums/status_effects/debuffs.dm

```diff
@@ -390,6 +390,15 @@
 			I.take_damage(100)
 	return ..()
 
+/// A status effect used for specifying confusion on a living mob.
+/// Created automatically with /mob/living/set_confusion.
+/datum/status_effect/confusion
+	id = "confusion"
+	alert_type = null
+
+/datum/status_effect/confusion/tick()
+	owner.set_confusion(owner.confused - 1)
```

Comment:
Check how the `/datum/status_effect/incapacitating` effects do it, and host the `confused` variable inside the datum, same as them.
Replicate procs such as `/mob/living/proc/AmountStun()` and the others to access the properties.